### PR TITLE
Adjust CheckedKeywordRecommender for checked user-defined operators.

### PR DESCRIPTION
**Pull request by AlekseyTs**
Test plan https://github.com/dotnet/roslyn/issues/59458

### CODE REVIEW

File: src/Features/CSharp/Portable/Completion/KeywordRecommenders/CheckedKeywordRecommender.cs [hunk 1]
```diff
@@ -18,10 +20,29 @@ public CheckedKeywordRecommender()
...
+                return true;
+            }
+
+            var targetToken = context.TargetToken;
+
+            if (targetToken.Kind() == SyntaxKind.OperatorKeyword)
+            {
+                if (targetToken.GetPreviousToken(includeSkipped: true).IsLastTokenOfNode<TypeSyntax>())
+                {
+                    return true;
```
Comment (jcouv) [inline src/Features/CSharp/Portable/Completion/KeywordRecommenders/CheckedKeywordRecommender.cs:36]: nit: consider adding comments to illustrate the situation (here and below)

**Comment by AlekseyTs** [comment]
@CyrusNajmabadi Please review changes past the first commit.

**Comment by AlekseyTs** [comment]
@CyrusNajmabadi Please review changes past the first commit.

**Comment by AlekseyTs** [comment]
@CyrusNajmabadi Please review changes made after the first commit.

**Comment by AlekseyTs** [comment]
@CyrusNajmabadi Please review changes made after the first commit.

File: src/Features/CSharp/Portable/Completion/KeywordRecommenders/CheckedKeywordRecommender.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/Features/CSharp/Portable/Completion/KeywordRecommenders/CheckedKeywordRecommender.cs:39]: note: i would prefer this as well.  but it's not a problem to not have it.

File: src/Features/CSharp/Portable/Completion/KeywordRecommenders/CheckedKeywordRecommender.cs [hunk 2]
```diff
@@ -18,10 +21,49 @@ public CheckedKeywordRecommender()
...
+
+                SyntaxToken previousToken;
+
+                if (previousPossiblySkippedToken.IsLastTokenOfNode<ExplicitInterfaceSpecifierSyntax>())
+                {
+                    var firstSpecifierToken = previousPossiblySkippedToken.GetAncestor<ExplicitInterfaceSpecifierSyntax>().GetFirstToken(includeSkipped: true);
+
+                    if (firstSpecifierToken.GetPreviousToken(includeSkipped: true).IsLastTokenOfNode<TypeSyntax>())
+                    {
+                        return true;
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/Completion/KeywordRecommenders/CheckedKeywordRecommender.cs:50]: so 'checked' is allowed, in an explicit-impl, even without having 'explicit' in the sig?

File: src/EditorFeatures/CSharpTest2/Recommendations/CheckedKeywordRecommenderTests.cs [hunk 3]
```diff
@@ -122,5 +122,172 @@ public async Task TestAfterRefExpression()
...
+@"class Goo {
+    public static explicit I1.operator $$ int (Goo x){}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestAfterOperator_ExplicitImplementation_01()
+        {
+            await VerifyKeywordAsync(
+@"class Goo { 
+    public static int I1.operator $$");
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest2/Recommendations/CheckedKeywordRecommenderTests.cs:258]: so this is legal from teh lang perspective?  it doesn't need to be marked as 'explicit'?

**Review by CyrusNajmabadi — APPROVED** [review]

File: src/Features/CSharp/Portable/Completion/KeywordRecommenders/CheckedKeywordRecommender.cs (hunk 2 above)
Comment (AlekseyTs) [reply src/Features/CSharp/Portable/Completion/KeywordRecommenders/CheckedKeywordRecommender.cs:50]: > so 'checked' is allowed, in an explicit-impl, even without having 'explicit' in the sig? 

An 'explicit' keyword has nothing to do with explicit interface implementation, it is part of a conversion operator declaration

Comment (CyrusNajmabadi) [reply src/Features/CSharp/Portable/Completion/KeywordRecommenders/CheckedKeywordRecommender.cs:50]: sorry, what i meant here is that this code checks if we're after an explicit-interface specifier, and then checks that right before that is a type-syntax.  But it does *not* do the `previousToken.Kind() == SyntaxKind.ExplicitKeyword` check below.  I'm trying to understand if that's ok and desirable behavior.  I commented on the test that seems to be hitting this, and i was trying to understand if this is a legal language construct and what it means.

File: src/EditorFeatures/CSharpTest2/Recommendations/CheckedKeywordRecommenderTests.cs (hunk 3 above)
Comment (AlekseyTs) [reply src/EditorFeatures/CSharpTest2/Recommendations/CheckedKeywordRecommenderTests.cs:258]: > so this is legal from teh lang perspective? it doesn't need to be marked as 'explicit'?

If we have a type before the specifier, this is not a conversion operator declaration. The 'explicit' keyword is used only for conversion operators.

File: src/Features/CSharp/Portable/Completion/KeywordRecommenders/CheckedKeywordRecommender.cs (hunk 2 above)
Comment (AlekseyTs) [reply src/Features/CSharp/Portable/Completion/KeywordRecommenders/CheckedKeywordRecommender.cs:50]: A conversion operator cannot have a type between 'explicit' and 'operator', its type goes before parameters. For example:
```
public static explicit operator checked TypeToConvertTo(TypeToConvertFrom x) {}
```

**Pull request merged**